streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -801,8 +801,11 @@ public void shouldWarnAboutRocksDBConfigSetterIsNotGuaranteedToBeBackwardsCompat
         props.setProperty(StreamsConfig.ROCKSDB_CONFIG_SETTER_CLASS_CONFIG, TestRocksDbConfigSetter.class.getName());
 
         final LogCaptureAppender appender = LogCaptureAppender.createAndRegister();
```

Comment:
Seems like this appender could also be scoped to the class that logs the message we want to capture, right?

Reply:
Unfortunately not; I tied it and it fails. The issue is, we would want to register the LogCaptureAppender only for `KafkaStreams.class` but at this point in the test, the corresponding logger does not exist yet: it's only created in the constructor call, that we actually test. Does this make sense?